src/actions/activity.js

```diff
@@ -3,22 +3,26 @@
 import { ACTIVITY_MESSAGES_LIMIT } from '../global/Constants';
 import { type ActivityLogMessage } from '../types/ActivityLogMessage';
 
+export const MESSAGE_ADDED = 'MESSAGE_ADDED';
+export const START_FETCH_MESSAGES = 'START_FETCH_MESSAGES';
+export const DONE_FETCH_MESSAGES = 'DONE_FETCH_MESSAGES';
+export const ADD_DUMMY_MESSAGE = 'ADD_DUMMY_MESSAGE';
+export const SHOW_SPINNER = 'SHOW_SPINNER';
```

Comment:
Activity state is aimed for handling activity logs for dashboard panel (sorry, there is no docs about that). Please move chat loading handling into separate chat actions/reducers files. 
Also, please use less UI-related names for logical actions to be independent from UI. For example START_CHAT_LOADING or CHAT_LOADING_STARTED instead of SHOW_SPINNER, etc.

Reply:
I was going to suggest a common way of handling the spinner.
We don't have to manage the spinner in every screen. We can create a common spinner handler and it can be used in every screen.
I will move the code into a new file actions/common and reducers/common.

Reply:
I'm not sure about how do you suggest to handle it in a common way, since each loading flag corresponds to different kind of requests and is independent from other loading flags.

Reply:
Dependency on screen is quite unstable. Two quick counter-examples:
 - Dashboard has several components that is loaded independently, so each of them should have different flag.
 - Nation indexing is started before any screen is showing, so once some screen is loaded, how can we decide if loading should be shown there?

I agree that adding inProgress flag is bad, however it's because its naming and goal is unclear. I would call it nationIndexingInProgress, for example. 
We may consider generalizing async action handling, since for each we have:
 - init action that starts operation
 - successful action callback
 - failed action callback
 - error field in state
 - progress field in state.
I guess we will think about it during refactor session.

Reply:
yes - I agree. We should handle this during the refactor session.

Reply:
Ok, I moved it to chat.js for now. We can handle it during the refactor session.